File: frontend/src/api/capabilities.ts
export type FeatureCapability = {
  available: boolean;
  reason: string | null;
};


export type Capabilities = {
  patients: FeatureCapability;
  viewer: FeatureCapability;
  anatomy: FeatureCapability;

  digital_twin: FeatureCapability;
  runs: FeatureCapability;
  tools: FeatureCapability;
  research: FeatureCapability;
};


export async function fetchCapabilities():
Promise<Capabilities> {
  const response = await fetch(
    "/api/capabilities",
  );

  if (!response.ok) {
    throw new Error(
      "Failed to load backend capabilities",
    );
  }

  return (
    await response.json()
  ) as Capabilities;
}

File: src/gbm_twin/api/schemas/capabilities.py
from __future__ import annotations

from pydantic import BaseModel


class FeatureCapability(BaseModel):
    available: bool
    reason: str | None = None


class CapabilitiesResponse(BaseModel):
    patients: FeatureCapability
    viewer: FeatureCapability
    anatomy: FeatureCapability

    digital_twin: FeatureCapability
    runs: FeatureCapability
    tools: FeatureCapability
    research: FeatureCapability